Clean up, renaming findKeyFromPubHash to findKeyFromPubKeyHash

dukeh3 opened the pull request:
Minor clean up, but I think its better that the interfaces are consistent across the different components.

So the hash for that pubKey should be named pubKeyHash not pubHash.
And the pubKey should be named pubKey, not pubkey.

msgilligan reviewed (changes requested):
Thanks for this. We absolutely should rename this for consistency. Thanks for keeping the old methods with the `@Deprecated` annotation. You also should add the `@deprecated` JavaDoc tag before the "Use xxx" comments.

For bitcoinj, we are very insistent upon an individual, incremental change (by PR or by commit) being limited to a single change. There should be no unrelated whitespace or formatting changes.  (Maybe a fix of a typo in a JavaDoc comment you are changing anyway, but nothing beyond that.) Not only does this make PRs easier to review, it results in a very clean Git history.

Thanks!

Commit:
Apply local patches on top of upstream sync

- Switch BouncyCastle from bcprov-jdk15to18 to bcprov-jdk18on (1.84)
- Add apostrophe notation support to ChildNumber.parse()
- Add HDPath test vectors for BIP-32 apostrophe notation
